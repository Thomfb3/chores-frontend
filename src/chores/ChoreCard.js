import React, { useContext } from "react";
import { Link } from "react-router-dom";
import formatDate from "../helpers/formatDate";
import findUserInTeam from "../helpers/findUserInTeam";
import UserContext from "../auth/UserContext";

function ChoreCard({
    id,
    title,
    description,
    points,
    status,
    assigneeUsername,
    assigneeName,
    assigneeImage,
    choreImage,
    assigner, 
    assignee,
    createdBy,
    createdAt, 
    dueDate,
    cardType 
    }) {

    console.debug("ChoreCard");
    const { currentUser, currentTeamUsers } = useContext(UserContext);

    if (cardType==="MyChoreList") {
        const userAssignee = findUserInTeam(assignee, currentTeamUsers);
        const userAssigner = findUserInTeam(assigner, currentTeamUsers);

        return (
            <Link to={`/chores/${id}`}>
                <div className="" style={{border: "1px solid red", margin: "20px"}}>
                <div className="">{userAssignee.profileImage && 
                <img 
                    className="" 
                    alt={userAssignee.profileImage}
                    src={userAssignee.profileImage}
                     /> 
                    }
                <p>{userAssignee.firstName}</p>
                </div>
                    <div className="">
                        <h4>{title}</h4>
                        <p>{description}</p>
                        <p className="">{points}</p>
                        <p className="">{formatDate(dueDate)}</p>
                        <p className="">{`Chore was created at ${formatDate(createdAt)}, by ${userAssigner.username}`}</p>
                    </div>
                </div>
            </Link>
        );

    } else if (cardType==="TeamChoresList") { 
        return (
            <Link to={`/chores/${id}`}>
                <div className="" style={{border: "1px solid red", margin: "20px"}}>
                <div className="">{assigneeImage && 
                <img 
                    className="" 
                    alt={assigneeImage}
                    src={assigneeImage}
                     /> 
                    }
                <p>{assigneeUsername}</p>
                <p>{assigneeName}</p>
                </div>
    
                    <div className="">
                        <h4>{title}</h4>
                        <p>{status}</p>
                        <p>{description}</p>
                        <p className="">{points}</p>
                        <p className="">{formatDate(dueDate)}</p>
                    </div>
                </div>
            </Link>
        );
    } else if (cardType==="UnclaimedChoresList") {
        const createdByUser = findUserInTeam(createdBy, currentTeamUsers);
        return (
            <Link to={`/chores/${id}`}>
                <div className="" style={{border: "1px solid red", margin: "20px"}}>
       
                    <div className="">
                        <h4>{title}</h4>
                        <p>{description}</p>
                        <p className="">{points}</p>
                        <p className="">{formatDate(dueDate)}</p>
                        <p className="">{`Chore was create at ${formatDate(createdAt)}, by ${createdByUser.username}`}</p>
                    </div>
                </div>
            </Link>
        );
    }
};


export default ChoreCard;




