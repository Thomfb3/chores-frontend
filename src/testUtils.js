import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

