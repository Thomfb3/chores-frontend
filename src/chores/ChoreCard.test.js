import React from "react";
import { render } from "@testing-library/react";
import ChoreCard from "./ChoreCard";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";


it("matches snapshot", function () {
  
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <ChoreCard
                    id="userid1"
                    key="userid1"
                    title="Clean Bathroom"
                    description="clean the bathroom"
                    points={200}
                    status="approved"
                    assigneeImage="defaultPic.jpeg"
                    choreImage="defaultChore.jpg"
                    assignerID="userid1"
                    assignee="userid1"
                    createdAt="2021-11-05T03:47:24.581+00:00"
                    createdBy="61837e4a9b875f020816b656"
                    currentUser={true}
                    dueDate="2021-11-05T02:24:00.000+00:00"
                    cardType="MyChoreList"
                />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
