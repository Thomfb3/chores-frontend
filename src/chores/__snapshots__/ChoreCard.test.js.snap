// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="ChoreCard"
    href="/chores/userid1"
  >
    <div
      class="ChoreCard__container ChoreCard__approved "
    >
      <div
        class="ChoreCard__inner-container"
      >
        <div>
          <p
            class="ChoreCard__title"
          >
            Clean Bathroom
          </p>
          <p
            class="ChoreCard__description"
          >
            clean the bathroom
          </p>
        </div>
        <div
          class="ChoreCard__points-section"
        >
          <p
            class="ChoreCard__points"
          >
            200 pts
          </p>
        </div>
        <div
          class="ChoreCard__due-date"
        >
          <div
            class="ChoreCard__approved-section ChoreCard__approved-section--my-chores"
          >
            <p>
              Approved
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gjr16u-MuiSvgIcon-root"
              data-testid="CheckCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <p
          class="ChoreCard__footer"
        >
          Chore was created and assigned to  by undefined at Thursday, November 4, 2021 - 11:47 PM. Submit chore by due date.
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
